core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1137,12 +1144,28 @@ private[spark] class DAGScheduler(
     }
 
     stage.makeNewStageAttempt(partitionsToCompute.size, taskIdToLocations.values.toSeq)
-
-    // If there are tasks to execute, record the submission time of the stage. Otherwise,
-    // post the even without the submission time, which indicates that this stage was
-    // skipped.
     if (partitionsToCompute.nonEmpty) {
+      // If there are tasks to execute, record the submission time of the stage. Otherwise,
+      // post the even without the submission time, which indicates that this stage was
+      // skipped.
       stage.latestInfo.submissionTime = Some(clock.getTimeMillis())
+
+      // While an indeterminate stage retried, the stage attempt id will be used to extend the
+      // shuffle file in shuffle write task, and then the mapping of shuffle id to indeterminate
+      // stage id will be used for shuffle reader task.
+      if (stage.latestInfo.attemptNumber() > 0 && stage.isIndeterminate) {
+        // deal with shuffle writer side property.
+        stage match {
+          case sms: ShuffleMapStage =>
+            val stageAttemptId = stage.latestInfo.attemptNumber()
```

Comment:
`stage.latestInfo.attemptNumber()` is called twice, we can create a variable first.

Reply:
Thanks, done in 4c4af33.